rocketpool-daemon/common/state/network-state.go

```diff
@@ -337,7 +337,8 @@ func CreateNetworkStateForNode(cfg *config.SmartNodeConfig, rp *rocketpool.Rocke
 		return nil, nil, fmt.Errorf("error getting Protocol DAO proposal details: %w", err)
 	}
 	logger.Info(fmt.Sprintf("%d/%d - Retrieved Protocol DAO proposals", currentStep, steps), slog.Duration(keys.TotalElapsedKey, time.Since(start)))
-	currentStep++
+	// If adding more steps below, don't forget to increment currentStep:
```

Comment:
Does this feel like a footgun? The `currentStep++` at the end wasn't hurting anything and prevents someone from messing it up in the future if they add more steps.

Reply:
ineffassign doesn't like nonfunctional value assignments to variables.

It is a footgun, because the abstraction is poor- the maintainer shouldn't need to remember to increment something when moving between steps.

Instead, they should formally define the steps in a slice, and they should execute in a `for` loop. That is, however, out of scope for this change. I'll file to follow-up.

Reply:
Ok, let's take it out for now and I'll await the follow-up refactor.